src/python/pants/backend/python/lint/docformatter/rules.py

```diff
@@ -50,10 +52,25 @@ async def docformatter_fmt(
             ),
             input_digest=request.snapshot.digest,
             output_files=request.files,
-            description=(f"Run Docformatter on {pluralize(len(request.files), 'file')}."),
+            description=description,
             level=LogLevel.DEBUG,
         ),
     )
+    # Docformatter 1.6.0+ very annoyingly returns an exit code of 3 if run with `--in-place`
+    # and any files changed. Earlier versions do not return this code in fmt mode.
+    # (All versions return 3 in check mode if any files would have changed, but that is
+    # not an issue here).
+    if result.exit_code not in [0, 3]:
+        # TODO(#12725):It would be more straightforward to force the exception with:
```

Comment:
I don't get the reference to #12725

Reply:
Ah, I cargo-culted that from other files that mentioned the inability to convert FallibleProcessResult to ProcessResult. 

Yeah, I'm not 100% sure why that is the referenced PR.

Reply:
also, I would expect the TODO to reference an open issue.. ;)